### PR TITLE
[Merged by Bors] - feat(Mathlib/Algebra/Order): add equivalent conditions to `a ≤ b`

### PR DESCRIPTION
**Pull request by yoh-tanimoto**
Two theorems that give equivalent conditions to `a ≤ b` in terms of `ε`-approximation.
- Generalize `a ≤ b ↔ ∀ ε, 1 < ε → a ≤ b * ε` (`le_iff_forall_one_lt_le_mul`) to monoids.
- Add `(hb : 0 ≤ b) : a ≤ b ↔ ∀ ε, 1 < ε → a ≤ b * ε` for linearly ordered semifields.

the proofs were suggested by @fpvandoorn in [this thread](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/.60le_iff_forall_pos_le_add.60.20for.20.60NNReal.60)

motivation: I will need them in the proof of regularity of `rieszContent`, that will be defined after #18775

### CODE REVIEW

File: Mathlib/Algebra/Order/Monoid/Unbundled/ExistsOfLE.lean [hunk 1]
```diff
@@ -82,4 +82,10 @@ theorem le_iff_forall_one_lt_lt_mul' [MulLeftStrictMono α] :
     a ≤ b ↔ ∀ ε, 1 < ε → a < b * ε :=
   ⟨fun h _ => lt_mul_of_le_of_one_lt h, le_of_forall_one_lt_lt_mul'⟩
 
+set_option linter.docPrime false
```
Comment (j-loreaux) [inline Mathlib/Algebra/Order/Monoid/Unbundled/ExistsOfLE.lean:85]: You should add a docstring here instead of disabling the linter.

Comment (fpvandoorn) [reply Mathlib/Algebra/Order/Monoid/Unbundled/ExistsOfLE.lean:85]: Also, you can try if `le_iff_forall_one_lt_le_mul` is provable by exactly `le_iff_forall_one_lt_le_mul'`. If so, we can remove that lemma and just use `le_iff_forall_one_lt_le_mul` for the name of this lemma.

**Review by fpvandoorn — APPROVED** [review]
LGTM besides the one comment

File: Mathlib/Algebra/Order/Monoid/Unbundled/ExistsOfLE.lean (hunk 1 above)
Comment (yoh-tanimoto) [reply Mathlib/Algebra/Order/Monoid/Unbundled/ExistsOfLE.lean]: I checked that with the same assumption of [le_iff_forall_one_lt_le_mul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Group/DenselyOrdered.html#le_iff_forall_one_lt_le_mul) `exact?` gives `le_iff_forall_one_lt_le_mul'`, so I removed the former and removed `'` from the latter. As `Mathlib.Algebra.Order.Monoid.Unbundled.ExistsOfLE` is imported in `Mathlib.Algebra.Order.Group.DenselyOrdered`, this should cause no problem

File: Mathlib/Algebra/Order/Field/Basic.lean [hunk 2]
```diff
@@ -435,6 +435,16 @@ theorem inv_pow_anti (a1 : 1 ≤ a) : Antitone fun n : ℕ => (a ^ n)⁻¹ := fu
 theorem inv_pow_strictAnti (a1 : 1 < a) : StrictAnti fun n : ℕ => (a ^ n)⁻¹ := fun _ _ =>
   inv_pow_lt_inv_pow_of_lt a1
 
+theorem le_iff_forall_one_lt_le_mul₀ {α : Type*} [LinearOrderedField α]
```
Comment (fpvandoorn) [inline Mathlib/Algebra/Order/Field/Basic.lean:438]:
```suggestion
theorem le_iff_forall_one_lt_le_mul₀ {α : Type*} [LinearOrderedSemifield α]
```
This was suggested on Zulip to also work.

Other than that, LGTM. (I edited your commit message btw)

bors d+

**Pull request closed without merging**